src/simulation/m_bubbles.fpp

```diff
@@ -171,21 +186,28 @@ contains
         !!  @param fV Current bubble velocity
         !!  @param fR0 Equilibrium bubble radius
         !!  @param fpbdot Time derivative of the internal bubble pressure
-    function f_Hdot(fCpbw, fCpinf, fCpinf_dot, fntait, fBtait, fR, fV, fR0, fpbdot)
+        !!  @param gam_l Physical Bubble parameter (pass in from global parameters)
+        !!  @param Ca_l Cavitation number (pass in from global parameters)
+        !!  @param Web_l Weber number (pass in from global parameters)
+        !!  @param Re_inv_l Inverse Reynolds number (pass in from global parameters)
+        !!  @param polytropic_l Polytropic  switch (pass in from global parameters)
+    pure elemental function f_Hdot(fCpbw, fCpinf, fCpinf_dot, fntait, fBtait, fR, fV, fR0, fpbdot, gam_l, Ca_l, Web_l, Re_inv_l, polytropic_l)
         !$acc routine seq
         real(wp), intent(in) :: fCpbw, fCpinf, fCpinf_dot, fntait, fBtait
         real(wp), intent(in) :: fR, fV, fR0, fpbdot
+        real(wp), intent(in) :: gam_l, Ca_l, Web_l, Re_inv_l
+        logical, intent(in) :: polytropic_l
```

Comment:
Similar to above, since these are `intent(in)` why was the refactor necessary to make it `pure`?